content/courses/advanced-ada/chapters/packages.rst

```diff
@@ -3,12 +3,465 @@ Packages
 
 .. include:: ../../global.txt
 
+.. _Adv_Ada_Package_Renaming:
+
+Package renaming
+----------------
+
+We've seen in the
+:doc:`Introduction to Ada course </courses/intro-to-ada/chapters/modular_programming>`
+that we can :ref:`rename packages <Package_Renaming>`.
+
+Grouping packages
+~~~~~~~~~~~~~~~~~
+
+A use-case that we haven't mentioned in that course is that we can apply
+package renaming to group individual packages into a common hierarchy. For
+example:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Packages.Package_Renaming_1
+
+    package Driver_M1 is
+
+    end Driver_M1;
+
+    package Driver_M2 is
+
+    end Driver_M2;
+
+    package Drivers
+      with Pure is
+
+    end Drivers;
+
+    with Driver_M1;
+
+    package Drivers.M1 renames Driver_M1;
+
+    with Driver_M2;
+
+    package Drivers.M2 renames Driver_M2;
+
+Here, we're renaming the :ada:`Driver_M1` and :ada:`Driver_M2` packages as
+child packages of the pure :ada:`Drivers` package. (We discuss more about pure
+packages :ref:`later <Adv_Ada_Pure>`.)
+
+
+Child of renamed package
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Note that we cannot create a child package for a package that has been renamed.
```

Comment:
The phrase "for a package that has been renamed" is a bit problematic because Driver_M1 and Driver_M2 are the "packages that have been renamed" (as Drivers.M1 and Drivers.M2). So perhaps it should read something like this:
"Note that we cannot create a child package using a parent package name that was introduced by a renaming."

Reply:
Yes, your version sounds better. Thanks!